modules/channel-web/src/views/lite/components/messages/Message.tsx

```diff
@@ -99,6 +99,15 @@ class Message extends Component<Renderer.Message> {
     return '*Unsupported message type*'
   }
 
+  handleContextMenu = e => {
+    const showContextMenu = window.botpress.extensions && window.botpress.extensions.showContextMenu
+    if (showContextMenu) {
+      const customActions = this.props.store.view.customActions
+      const id = this.props.incomingEventId
+      showContextMenu(e, { id, customActions })
+    }
+  }
+
```

Comment:
What do you think of passing the complete props to showContextMenu? In this case we use incomingEventId, but it may be possible that we need another of the available prop

Reply:
I believe the day we'll need other informations it will be super easy to do the refactor. I think we should only pass the strict necessary.

Reply:
But if you return the props instead of props.incomingEventId, that refactor won't be necessary and will scale much easier

Reply:
No I really don't think the context menu should receive all the props object. It's none of his business. Modularity is not the only quality of a well built code, separation of concern also is.

Reply:
There's no real room for discussion here, That's juste a matter of preferences. while you're right that at the moment it's not of its concerns but it might become and once the times come it would indeed be easier for the dev to use the desired prop without changing the method signature.

It doesn't had ay overhead to pass the whole object and it ensures we don't make breaking changes in the future.